src/exercises/day-1/solutions-morning.md

```diff
@@ -7,3 +7,12 @@
 ```rust
 {{#include for-loops.rs}}
 ```
+### Bonus question
+
+A two-dimensional slice-of-slices would be fine for argument if you do something like
```

Comment:
This is where I suggest you expand on the problems — it honestly doesn't work so well.

Something like

> It might seem that we could use a slice-of-slices (`&[&[i32]]`) as the input type to `transpose` and thus make our function handle any size of matrix. However, this quickly breaks down: the return type cannot be `&[&[i32]]` since it needs to own the data you return.
>
> You can attempt to use something like `Vec<Vec<i32>>`, but this doesn't work very well either: it's hard to convert from `Vec<Vec<i32>>` to `&[&[i32]]` so now you cannot easily use `pretty_print` either.